tests/table/hovered_table_store.test.ts

```diff
@@ -0,0 +1,86 @@
+import { Model, UID } from "../../src";
+import { HoveredTableStore } from "../../src/components/tables/hovered_table_store";
+import { TABLE_HOVER_BACKGROUND_COLOR } from "../../src/constants";
+import { createTable, setCellContent } from "../test_helpers/commands_helpers";
+import { makeStore } from "../test_helpers/stores";
+
+describe("Hovered Table Store", () => {
+  let hovered_table_store: HoveredTableStore;
+  let model: Model;
+  let sheetId: UID;
+
+  beforeEach(() => {
+    ({ model, store: hovered_table_store } = makeStore(HoveredTableStore));
+    hovered_table_store.clear();
+    sheetId = model.getters.getActiveSheetId();
+    createTable(model, "A1:B2");
+  });
+
+  test("Should not store overlay for header cells, regardless of data", () => {
+    const headerPositionNoData = { col: 0, row: 0 };
+    model.updateMode("dashboard");
+    hovered_table_store.hover(headerPositionNoData);
+    expect(hovered_table_store.overlayColors.has({ sheetId, ...headerPositionNoData })).toBe(false);
+
+    const headerPositionWithData = { col: 0, row: 0 };
+    setCellContent(model, "A1", "Header");
+    hovered_table_store.hover(headerPositionWithData);
+    expect(hovered_table_store.overlayColors.has({ sheetId, ...headerPositionWithData })).toBe(
+      false
+    );
+  });
+
+  test("Should store overlay for full data rows with content in dashboard mode", () => {
+    const position = { col: 0, row: 1 };
```

Comment:
you can even name the position :) (same below)
```suggestion
    const A2 = { col: 0, row: 1 };
```